hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlUnionTest.java

```diff
@@ -240,19 +242,122 @@ public void watermarkedStreamUnionAllTest() {
         ResultIterator<SqlRow> iterator = (ResultIterator<SqlRow>) result.iterator();
         List<Row> actualRows = new ArrayList<>();
         assertTrueEventually(() -> {
-            while (iterator.hasNext(50, TimeUnit.MILLISECONDS) == HasNextResult.YES) {
-                actualRows.add(new Row(iterator.next()));
-            }
-            if (actualRows.size() == 7 && expected.size() < 7) {
-                // Because we drop late items after union all (thanks to UnionDropLateItemsTransposeRule),
-                // it can happen that one of the late items makes it through. But both will never make it.
-                expected.add(new Row(timestampTz(0L), 4));
-            }
-            assertThat(actualRows).containsExactlyInAnyOrderElementsOf(expected);
-        },
+                    while (iterator.hasNext(50, TimeUnit.MILLISECONDS) == HasNextResult.YES) {
+                        actualRows.add(new Row(iterator.next()));
+                    }
+                    if (actualRows.size() == 7 && expected.size() < 7) {
+                        // Because we drop late items after union all (thanks to UnionDropLateItemsTransposeRule),
+                        // it can happen that one of the late items makes it through. But both will never make it.
+                        expected.add(new Row(timestampTz(0L), 4));
+                    }
+                    assertThat(actualRows).containsExactlyInAnyOrderElementsOf(expected);
+                },
                 5);
     }
 
+    @Test
+    public void prunableUnionAllTest() {
+        instance().getConfig().addMapConfig(
+                new MapConfig("pMap1").setPartitioningAttributeConfigs(List.of(
+                        new PartitioningAttributeConfig("id")
+                )));
+        instance().getConfig().addMapConfig(
+                new MapConfig("pMap2").setPartitioningAttributeConfigs(List.of(
+                        new PartitioningAttributeConfig("id")
+                )));
+
+        IMap<Person, String> prunableMap1 = instance().getMap("pMap1");
+        IMap<Person, String> prunableMap2 = instance().getMap("pMap2");
+
+        createMapping("pMap1", Person.class, String.class);
+        createMapping("pMap2", Person.class, String.class);
+
+        for (int i = 0; i < 5; ++i) {
+            prunableMap1.put(new Person(i, "ABC" + i), "" + i);
+            prunableMap2.put(new Person(i, "ABC" + i), "" + i);
+        }
+
+        String sql = "(SELECT this FROM pMap1 WHERE id = 1) UNION ALL (SELECT this FROM pMap2 WHERE id = 1 OR id = 2)";
```

Comment:
second subquery is not prunable

Reply:
however this test demonstrates that duplicates are not lost which is good to test in case of partition pruning

Reply:
Yes, I will fix it, but keep the test itself.